frontend/src/pages/articles/components/article-filters/article-filters.tsx

```diff
@@ -0,0 +1,97 @@
+import { matchPath } from 'react-router-dom';
+
+import { Input } from '~/libs/components/components.js';
+import { Select } from '~/libs/components/select/select.js';
+import { AppRoute, ArticleSubRoute } from '~/libs/enums/enums.js';
+import {
+  useAppForm,
+  useCallback,
+  useDeepCompareEffect,
+  useLocation,
+} from '~/libs/hooks/hooks.js';
+import { type SelectOption } from '~/libs/types/select-option.type.js';
+
+import { type FilterFormValues } from '../../libs/types/types.js';
+import { DEFAULT_FILTER_PAYLOAD } from './libs/constants/constants.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  genreSelectOptions: SelectOption[];
+  authorsSelectOptions: SelectOption[];
+  onSubmit: (payload: FilterFormValues) => void;
+};
+
+const ArticleFilters: React.FC<Properties> = ({
+  genreSelectOptions,
+  authorsSelectOptions,
+  onSubmit,
+}) => {
+  const { control, errors, watch, handleReset } = useAppForm<FilterFormValues>({
+    defaultValues: DEFAULT_FILTER_PAYLOAD,
+    mode: 'onChange',
+  });
+  const { pathname } = useLocation();
+  const isMyArticlesPage = matchPath(
+    { path: `${AppRoute.ARTICLES}/${ArticleSubRoute.MY_ARTICLES}` },
+    pathname,
+  );
+  const data: FilterFormValues = watch();
+
+  useDeepCompareEffect(() => {
+    const handler = setTimeout(() => {
```

Comment:
I guess it would be better to name this variable as `timerId` instead

Reply:
What variable ? handler ? it's a form state change handler so I think name is fine as ti is.  Or you were talking about something else ?

Reply:
I mean `handler` variable, actually `setTimeout` returns a timer id, not the handler you pass into it, so the name is a little bit confusing in my opinion 😅

Reply:
I know that's technically it's a timer id but semantically it represents code that will be executed after certain amount of time. At least in my opinion